erpnext/patches/v10_0/deduplicate_offline_pos.py

```diff
@@ -0,0 +1,26 @@
+import frappe
+
+
+def execute():
+    duplicates = frappe.db.sql(
+        'select name, offline_pos_name from `tabSales Invoice` where '
+        'offline_pos_name is not null group by '
+        'offline_pos_name having count(*) > 1'
+    )
+
+    for duplicate in duplicates:
+        keep, offline_pos_name = duplicate
+        duplicate_list = frappe.db.sql(
+            'select name from `tabSales Invoice` where '
+            'offline_pos_name=%s',
+            offline_pos_name
+        )
+
+        counter = 0
+        for item in duplicate_list:
+            name, = item
+            if name != keep:
+                si = frappe.get_doc('Sales Invoice', name)
+                counter += 1
+                new_name = '{0}-{1}'.format(si.offline_pos_name, counter)
+                si.db_set('offline_pos_name', new_name)
```

Comment:
Can you provide some information about this patch, based on code it seems that you are

Reply:
This patch will do the following:
- check for invoices that have the same offline_pos_name. This will give us a list of invoices which have been duplicated.
- If there are any duplicates, append a number to them. e.g, we find invoices SINV-0001 and SINV-0002 have the same offline POS name of 1234. SINV-0001 offline pos name remains as 1234 and SINV-0002 becomes 1234-1